src/externalDevice/BleChessProtocol.ts

```diff
@@ -1,22 +1,118 @@
 import { BaseProtocol, BaseState } from './BaseProtocol'
-import { isUciWithPromotion, isCentralStateCreated, createFullFen, lastMoveToUci, getCommandParams, sendCommandToPeripheral, sendMoveToCentral, areFensSame, sendStateChangeToCentral, applyPeripheralMoveRejected, applyPeripheralLastMove, applyPeripheralSynchronized, applyPeripheralPieces } from './utils'
+import { isCentralStateCreated, createFullFen, lastMoveToUci, getCommandParams, sendCommandToPeripheral, sendMoveToCentral, sendStateChangeToCentral, applyPeripheralMoveRejected, applyPeripheralLastMove, applyVariantSupported, applyPeripheralSynchronized, applyPeripheralPieces, createIterator } from './utils'
 import { State, makeDefaults } from '../chessground/state'
+import { GameStatus } from '../lichess/interfaces/game'
 import { Toast } from '@capacitor/toast'
 import i18n from '../i18n'
 
 export class BleChessProtocol extends BaseProtocol {
   roundState = makeDefaults()
-  features = new BleChessFeatures
+  features = new Features
+  variants = new Variants
+  variantsMap = {
+    standard: this.variants.standard,
+    chess960: this.variants.chess960,
+    antichess: this.variants.antiChess,
+    kingOfTheHill: this.variants.kingOfTheHill,
+    threeCheck: this.variants.threeCheck,
+    atomic: this.variants.atomic,
+    horde: this.variants.horde,
+    racingKings: this.variants.racingKings,
+    crazyhouse: this.variants.crazyHouse,
+  }
+  endReasonsMap = {
+    mate: EndReason.Checkmate,
+    stalemate: EndReason.Stalemate,
+    draw: EndReason.Draw,
+    timeout: EndReason.Timeout,
+    outoftime: EndReason.Timeout,
+    resign: EndReason.Resign,
+    aborted: EndReason.Abort,
+    noStart: EndReason.Undefined,
+    unknownFinish: EndReason.Undefined,
+    cheat: EndReason.Undefined,
+    variantEnd: EndReason.Undefined,
+  }
 
   init(st: State) {
     this.roundState = st
     this.transitionTo(new Init)
   }
 }
 
-class BleChessFeatures {
-  msg: boolean = false
-  lastMove: boolean = false
+enum FeatureName {
+  LastMove = 'last_move',
+  Check = 'check',
+  Msg = 'msg',
+}
+
+enum VariantName {
```

Comment:
shall we reuse defined in blichess imp?
e.g.  
game: {
    supportedVariants: ['standard', 'chess960', 'antichess', 'fromPosition',
      'kingOfTheHill', 'threeCheck', 'atomic', 'horde', 'racingKings', 'crazyhouse'
    ],
src/settings.ts

Reply:
blichess has `cameCase` notation, protocol allow only `under_score` notations

Reply:
Don't know how to handle `fromPosition` variant and if we should